technic/machines/register/cables.lua

```diff
@@ -43,84 +43,108 @@ local function check_connections(pos)
-						table.insert(network.RE_nodes, pos)
-					elseif eu_type == technic.producer_receiver then
-						table.insert(network.PR_nodes, pos)
-						table.insert(network.RE_nodes, pos)
-					elseif eu_type == technic.battery then
-						table.insert(network.BA_nodes, pos)
-					end
-					-- Note: SPECIAL (i.e. switching station) is not traversed!
-				end
-			elseif dead_end and not placed then
-				-- Dead end removed, remove it from the network
-				-- Get the network
-				local network_id = technic.cables[minetest.hash_node_position(positions[1])]
-				if not network_id then
-					-- We're evidently not on a network, nothing to add ourselves to
-					return
+				table.insert(network.all_nodes,pos)
+			elseif technic.machines[tier][node.name] then
+				-- Found a machine
+				local eu_type = technic.machines[tier][node.name]
+				meta:set_string(tier.."_network", string.format("%.20g", network_id))
+				meta:set_int(tier.."_EU_timeout", 2)
+				if     eu_type == technic.producer then
+					table.insert(network.PR_nodes, pos)
+				elseif eu_type == technic.receiver then
+					table.insert(network.RE_nodes, pos)
+				elseif eu_type == technic.producer_receiver then
+					table.insert(network.PR_nodes, pos)
+					table.insert(network.RE_nodes, pos)
+				elseif eu_type == technic.battery then
+					table.insert(network.BA_nodes, pos)
 				end
-				local network = technic.networks[network_id]
+				-- Note: SPECIAL (i.e. switching station) is not traversed!
+			end
+		else
+			-- Dead end removed, remove it from the network
+			-- Get the network
+			local network_id = technic.cables[minetest.hash_node_position(positions[1])]
+			if not network_id then
+				-- We're evidently not on a network, nothing to add ourselves to
+				return
+			end
+			local network = technic.networks[network_id]
 
-				-- Search for and remove machine
-				technic.cables[minetest.hash_node_position(pos)] = nil
-				for tblname,table in pairs(network) do
-					if tblname ~= "tier" then
-						for machinenum,machine in pairs(table) do
-							if machine.x == pos.x
-							and machine.y == pos.y
-							and machine.z == pos.z then
-								table[machinenum] = nil
-							end
+			-- Search for and remove machine
+			technic.cables[minetest.hash_node_position(pos)] = nil
+			for tblname,table in pairs(network) do
+				if tblname ~= "tier" then
+					for machinenum,machine in pairs(table) do
+						if machine.x == pos.x
+						and machine.y == pos.y
+						and machine.z == pos.z then
+							table[machinenum] = nil
 						end
 					end
 				end
-			else
-				-- Not a dead end, so the whole network needs to be recalculated
-				for _,v in pairs(technic.networks[net].all_nodes) do
-					local pos1 = minetest.hash_node_position(v)
-					technic.cables[pos1] = nil
-				end
-				technic.networks[net] = nil
 			end
 		end
+		return
+	end
+
+	local net = technic.cables[minetest.hash_node_position(pos)]
+	if net and technic.networks[net] then
+		clear_network(net)
+	end
```

Comment:
The name `net` does not clarify the variable contents. Furthermore, `t[nil]` cannot be assigned, thus the condition can be abbreviated:
```suggestion
	local network_id = technic.cables[minetest.hash_node_position(pos)]
	if technic.networks[network_id] then
		clear_network(network_id)
	end
```
.. or maybe have a guard in `clear_network` so that you can pass it directly without checking here.